### PR TITLE
Deduplicate BinderInfo/BinderInfoCC

### PR DESCRIPTION
**Pull request by bts**
/cc @joelburget @cartazio

* Deduplicate implementations
* Move to `Hopper.Internal.Type.BinderInfo`

### CODE REVIEW

File: src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs [hunk 1]
```diff
@@ -41,9 +40,9 @@ import qualified Data.Map as Map-- FIXME, use IntMap or WordMap
 import Hopper.Internal.Core.Literal
 import GHC.Generics
 import qualified  Data.Vector as V
-import Hopper.Internal.Type.Relevance(Relevance)
```
Comment (joelburget) [inline src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs]: Note: I prefer this formatting.

File: src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs [hunk 2]
```diff
@@ -41,9 +40,9 @@ import qualified Data.Map as Map-- FIXME, use IntMap or WordMap
...
+import Hopper.Internal.Type.BinderInfo (BinderInfo)
```
Comment (joelburget) [inline src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs]: Note: I prefer this formatting

File: src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs [hunk 3]
```diff
@@ -79,39 +78,9 @@ newtype CodeArity =
...
-data BinderInfoCC =
-      BinderInfoDataCC
-        { relevanceBICC :: Relevance -- if zero, during evaluation we need not pass around
-                                     --  BUT during NORMALIZATION, we probably do
-                                     --  so the normalizer WILL thread around irrelevant values
-                                     -- to properly support dependent type checking
-                                     -- as is proper, because a runtime irrelevant value
-                                     -- SHOULD be relevant during type checking, or
-                                     -- it has no reason to exist
-        , typeBICC :: TypeCC  -- at least for now, closure converted stuff may need a
```
Comment (joelburget) [inline src/Hopper/Internal/LoweredCore/ClosureConvertedANF.hs]: What happened to CC'ed stuff needing a "slightly different type system fragment"?

File: src/Hopper/Internal/Type/BinderInfo.hs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+data BinderInfo =
+  BinderInfoData { relevanceBI :: Relevance
+                 -- ^ if zero, during evaluation we need not pass around
+                 --  BUT during NORMALIZATION, we probably do
+                 --  so the normalizer WILL thread around irrelevant values
+                 -- to properly support dependent type checking
+                 -- as is proper, because a runtime irrelevant value
+                 -- SHOULD be relevant during type checking, or
+                 -- it has no reason to exist
+                 , typeBICC :: Type
```
Comment (joelburget) [inline src/Hopper/Internal/Type/BinderInfo.hs]: I guess `BinderInfoData` may eventually be paramaterized by the type?

**Comment by bts** [comment]
@cartazio any comments?

File: src/Hopper/Internal/Type/BinderInfo.hs (hunk 4 above)
Comment (cartazio) [inline src/Hopper/Internal/Type/BinderInfo.hs]: and/or we have that each "lowering" is a subset of the types in more restricted forms and maybe a few more extra assumed/builtin datatypes that are per se definable in userland

**Pull request merged**